src/javascript/_cookies.js

```diff
@@ -0,0 +1,101 @@
+(function (window) {
+  var $ = window.$;
+  var Cookies = window.Cookies;
+  if (!Cookies || !$) {
+    return;
+  }
+
+  var COOKIE_NAMES = {
+    PREFERENCES_SET: 'cookies_preferences_set',
+    POLICY: 'cookies_policy'
+  };
+  
+  var DEFAULT_POLICY = {
+    essential: true,
+    settings: false,
+    usage: false
+  };
+
+  var GovUKCookie = {
+    get: function (name) {
+      var value = Cookies.get(name);
+      if (value) {
+        return JSON.parse(value);
+      }
+      return value;
+    },
+    set: function (name, value) {
+      var GOVUK_COOKIE_OPTIONS = {
+        expires: 365 // days
+      };
+    
+      return Cookies.set(
+        name,
+        value,
+        GOVUK_COOKIE_OPTIONS
+      );
+    }
+  };
+
+  GovUKCookie.set(
+    COOKIE_NAMES.POLICY,
+    DEFAULT_POLICY
+  );
+
+  var $cookieBanner = $('.global-cookie-message-dfe-sign-in');
+  var $cookieAcceptButton = $cookieBanner.find('button.cookie-accept');
+
+  var getAcceptedAllPolicy = function () {
+    var acceptedPolicy = $.extend(DEFAULT_POLICY, {}); // deep copy
+
+    $.each(acceptedPolicy, function (key) {
+      acceptedPolicy[key] = true;
+    });
+  };
+
+  var onCookieAccept = function (event, newPolicy) {
+    var acceptedPolicy = newPolicy || getAcceptedAllPolicy();
+
+    GovUKCookie.set(
+      COOKIE_NAMES.POLICY,
+      acceptedPolicy
+    );
+
+    GovUKCookie.set(
+      COOKIE_NAMES.PREFERENCES_SET,
+      true
+    );
+
+    if (event.target === $cookieAcceptButton[0]) {
+      $cookieBanner.slideUp();
+    }
+  };
+
+  if (!GovUKCookie.get(COOKIE_NAMES.PREFERENCES_SET)) {
+    $cookieBanner.slideDown(function () {
+      $cookieAcceptButton.click(onCookieAccept);
+    });
+  }
+  
+  var $preferencesForm = $('.cookies-page-dfe-sign-in__preferences-form');
+  var $submitPrefsButton = $('#dsi-set-cookie-preferences');
+
+  $submitPrefsButton.on('click', function (event) {
+    event.preventDefault();
+    var newPolicy = {
+      settings: !!$preferencesForm.find("input[name='cookie.settings']:checked").val(),
+      usage: !!$preferencesForm.find("input[name='cookie.usage']:checked").val()
+    }
+
+    var acceptedPolicy = $.extend(DEFAULT_POLICY, newPolicy);
+    onCookieAccept(event, acceptedPolicy);
+
+    if (window.history) {
```

Comment:
would it not be better to just make the banner not visible?

Reply:
this code/line refers to the cookie preferences page and navigating away afterwards. Though, I think after some thought, I'll just add a message a flash message a bit like how gov.uk works, that'll be simpler

Reply:
on second thought... this is how they want it from acceptance criteria @joseluisgraa 
https://2z36g8.axshare.com/sign_in_page.html#Cookie_state=1&CSUM=1

Reply:
I don't have access to that page I'm afraid

Reply:
Oh I see that was only for the separate page where the user specifies cookie prefs. Fine with it then!